File: python/Easy/remove_element.py
puzzle = 'https://leetcode.com/problems/remove-element/description/'

class Solution:
    def removeElement(self, nums: list[int], val: int) -> int:
        i = 0
        while i < len(nums):
            if nums[i] == val:
                del nums[i]
            else:
                i += 1

# Runtime: 46ms beats 76.4%
# Memory: 16.3mb beats 72.72%

File: java/easy/removeElement.java
// Java doesn't have array methods like js and python to easily slice or remove an element.
// So I took the two pointer approach to reassign values of the array using the left and right indexes as I loop through.

class Solution {
  public int removeElement(int[] nums, int val) {
      int left = 0;

      for (int right = 0; right < nums.length; right++) {
          if (nums[right] != val) {
              nums[left] = nums[right];
              left++;
          };
      };

      return left;
  }
};